File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/Person.scala
package com.xebia.functional.xef.scala.auto

import com.xebia.functional.xef.scala.auto.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class Person(name: String, age: Int) derives SerialDescriptor, Decoder

@main def runPerson: Unit =
  conversation {
    val person = prompt[Person](Prompt("Hello made-up person, what is your name and age?"))
    println(s"Hello ${person.name}, you are ${person.age} years old.")
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/Person.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.OpenAI
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable data class Person(val name: String, val age: Int)

suspend fun main() {
  OpenAI.conversation {
    val person: Person = prompt("What is your name and age?")
    println("Hello ${person.name}, you are ${person.age} years old.")
  }
}
